handler.go

```diff
@@ -84,11 +88,16 @@ func (mux *Mux) Handle(ctx *Context, res *http.Response, err error) {
 	} else {
 		// Find a matching response handler
 		var h Handler
+		var criteria = -1
 		var n = -1
 		for r := range mux.res {
-			if ok, cnt := r.match(res); ok {
+			if ok, cnt, crit := r.match(res); ok {
 				if cnt > n {
-					h, n = r.h, cnt
+					h, crit, n = r.h, crit, cnt
```

Comment:
Shouldn't that assign to `criteria` here, instead of the if-scoped `crit` variable? i.e.:
```suggestion
					h, criteria, n = r.h, crit, cnt
```